Assets/_Laz/Scripts/Environment/Puzzles/Connection/ConnectingActivationBehaviour.cs

```diff
@@ -0,0 +1,121 @@
+using System.Collections.Generic;
+using PerigonGames;
+using Shapes;
+using UnityEngine;
+
+namespace Laz
+{
+    public class ConnectingActivationBehaviour : BasePuzzleBehaviour
+    {
+        [SerializeField] private NodeConnectionBehaviour[] _nodeBehaviours = null;
+        [SerializeField] private Polyline _line = null;
+        
+        private Queue<Edge> _listOfEdges = new Queue<Edge>();
+        private Queue<Vector3> _listOfNodePositions = new Queue<Vector3>();
+
+        public override void Initialize()
+        {
+            SetupQueueOfEdges();
+        }
+
+        public override void CleanUp()
+        {
+            base.CleanUp();
+            foreach (var nodeBehaviour in _nodeBehaviours)
+            {
+                nodeBehaviour.CleanUp();
+            }
+            _listOfEdges.Clear();
+            _listOfNodePositions.Clear();
+            _line.points = new List<PolylinePoint>();
+        }
+
+        public override void Reset()
+        {
+            base.Reset();
+            foreach (var nodeBehaviour in _nodeBehaviours)
+            {
+                nodeBehaviour.Reset();
+            }
+            _line.points = new List<PolylinePoint>();
+            SetupQueueOfEdges();
+        }
+
+        private void SetupQueueOfEdges()
+        {
+            var numberOfNodes = _nodeBehaviours.Length;
+            var arrayOfNodes = CreateArrayOfNodes(numberOfNodes);
```

Comment:
Up to you but I think its fine to just not cache arrayOfNodes variable

Reply:
sure